src/faking.ts

```diff
@@ -0,0 +1,275 @@
+ * })
+ */
+
+/*
+ * Irritatingly, Javascript does not provide a seedable RNG.  A seedable RNG is very
+ * helpful when you want reproduceable results.
+ */
+interface Xorshift128pState {
+    a: bigint,
+    b: bigint,
+}
+const MAX = (BigInt(1) << BigInt(64)) - BigInt(1);
+export function xorshift128p(s: Xorshift128pState): { state: Xorshift128pState, value: number } {
+    let t: bigint = s.a;
+    t ^= (t << BigInt(23)) & MAX;
+    t ^= (t >> BigInt(17));
+    t ^= s.b ^ (s.b >> BigInt(26));
+    const state = { a: s.b, b: t };
+    const value = Number((t + s.a) & MAX) / Number(MAX);
+    return { state, value };
+}
+
+/*
+ * Now obviously, I'm a monster for introducing monads to this.
+ * Here's a state monad. I'll explain why later.
+ */
+class State<T, V> {
+    runState: (s) => { state: T, value: V }
+
+    constructor(runState) {
+        this.runState = runState;
+    }
+
+    static pure<T, V2>(value: V2): State<T, V2> {
+        return new State((state: T) => ({ state, value }));
+    }
+
+    static get<T>(): State<T, T> {
+        return new State((state: T) => ({ state, value: state }));
+    }
+
+    static put<T>(s: T): State<T, undefined> {
+        return new State((state: T) => ({ state: s, value: undefined }));
+    }
+
+    static sequence<T, V>(a: State<T, V>[]): State<T, V[]> {
+        return a.reduce((acc, s) => acc.flatMap((vs) => s.map((v) => [...vs, v])),
+        State.pure([]));
+    }
+
+    static obj<T, V>(o: Record<string, State<T, any>>): State<T, Record<string, any>> {
+        return State.sequence(Object.entries(o).map(([k, s]) => s.map((v) => [k, v])))
+                    .map((Object as unknown as any).fromEntries);
+    }
+
+    map<V2>(f: (v: V) => V2): State<T, V2> {
+        return new State((s: T) => {
+            let { state, value } = this.runState(s);
+            return { state, value: f(value) };
+        });
+    }
+
+    flatMap<V2>(f: (v: V) => State<T, V2>): State<T, V2> {
+        return new State((s: T) => {
+            let { state, value } = this.runState(s);
+            return f(value).runState(state);
+        });
+    }
+}
+
+/*
+ * Using the state monad here allows us to write random object generators.
+ * We can build them up combinatorially while threading the state of
+ * the RNG through stuff.
+ *
+ * A state monad is basically a big, composed function that says "when you get this input
+ * state, do these things and return the new state and result".  The magic is that we can
+ * put them together, with helper functions that mutate the state, without ever having to
+ * think about threading the state around again.
+ *
+ * Normal, healthy programmers would probably just make it a global.
+ */
+
+ /* Helpfully, `xorshit128p` already follows the { state, value } pair output format that
+  * our state monad defines, so the implemntation is trivial */
+const Random = new State<Xorshift128pState, number>((s: Xorshift128pState) => xorshift128p(s));
```

Comment:
we don't use UpperCamelCase for variable names. If they're supposed to be class representations, they'll need to be classes.